testsuite/src/it/java/org/graalvm/tests/integration/DebugSymbolsTest.java

```diff
@@ -410,7 +411,10 @@ public static void carryOutGDBSession(StringBuffer stringBuffer, GDBSession gdbS
                                 Thread.sleep(100);
                             } while (gotoException != null && tryCount > 0);
                             if (gotoException != null) {
-                                fail("Unexpected GOTO failure: ", gotoException);
+                                errorQueue.add("Unexpected GOTO URL " + cp.c.split("URL ")[1] + " failure: " + gotoException.fillInStackTrace().toString());
+                            }
+                            if (MAX_GOTO_TRIES - tryCount < MAX_GOTO_TRIES - 2 && Commands.FAIL_ON_PERF_REGRESSION) {
```

Comment:
Why this magic `- 2`? How about simplifying this condition to `tryCount > 2` and replace `2` with a constant?

Reply:
Makes sense, Will amend. 

I am not sure what is reasonable here, hence magic numbers experimentation.
What would you suggest? 
The breakpoint should be set already at that time (slow). So why hitting it is so much slower now than before? With 100ms we are already at 100x :-D of what it would take to serve the page without debug session going on.

My point is not to nitpick on milliseconds. I am genuinely worried that if a giant timeout of 200m is not enough and we need 500ms+ here for such a test app, it actually might become unusable for a production, big app where attaching to a live system and clicking through app would cause browser to time out before hitting the breakpoint....

Reply:
Not sure if debugging and attaching to production workloads is the primary goal of the debug info work. IMO it's more for expert developers trying to fix a bug in native code (outside production).